### PR TITLE
Zero out client state before processing upgraded client proof

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #292
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/light-clients/07-tendermint/types/upgrade_test.go [hunk 1]
```diff
@@ -452,9 +452,6 @@ func (suite *TendermintTestSuite) TestVerifyUpgrade() {
 		cs := suite.chainA.GetClientState(path.EndpointA.ClientID)
 		clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), path.EndpointA.ClientID)
 
-		// Call ZeroCustomFields on upgraded clients to clear any client-chosen parameters in test-case upgradedClient
```
Comment (chatton) [inline modules/light-clients/07-tendermint/types/upgrade_test.go:455]: this caused a test an existing test case to pass `unsuccessful upgrade: committed client does not have zeroed custom fields`, this test now passes due to the zeroing in `VerifyUpgradeAndUpdateState`

File: modules/light-clients/07-tendermint/types/upgrade.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ import (
 // in client state that must be the same across all valid Tendermint clients for the new chain.
```
Comment (charleenfei) [inline modules/light-clients/07-tendermint/types/upgrade.go:18]: maybe add a comment that zeroing out the submitted client prevents the proposal from containing information governance is not actually voting on, see previous CHANGELOG comment

**Review by charleenfei — APPROVED** [review]
i would add some document the reasoning behind this change otherwise lgmt :)

**Review by damiannolan — COMMENTED** [review]
Should this also be targeted for `02-client-refactor` branch?

**Comment by chatton** [comment]
@damiannolan changing the base branch introduced a massive amount of merge conflicts, I think we can leave this PR open and once @colin-axner returns we can discuss if it needs to be included in the client refactor or if we can just merge to main.

**Comment by chatton** [comment]
Closing in favour of: #1674

**Pull request closed without merging**